### DISCUSSION TITLE
One of three nodes revision inconsistency cause the node compaction abnormal

### DISCUSSION THREAD

**Discussion by xrkCode** [etcd-io/etcd#16304] [Q&A]
### Bug report criteria

- [ ] This bug report is not security related, security issues should be disclosed privately via security@etcd.io.
- [ ] This is not a support request, support requests should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [x] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [x] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

1、one node db size is full, achieve 2g,  the log is
_ util.go:129: Warning: failed to apply request "header:<ID:3842561065849928906 username:\"" auth_revision:1 > put:<key:\"" value_size:313 lease:3842561065759985782 >" with response "size:8" took (147.544µs) to execute, err is etcdserver: no space_

2、the compact is abnormal，the log is
_util.go:129: Warning: failed to apply request "header:<ID:3842561065849944060 > compaction:<revision:385317169 > " with response "" took (9.894µs) to execute, err is mvcc: required revision is a future revision_

3、the revision of three node is inconsistency，two of the three is same,  the remaining abnormal node is different，In addition, the mod_revision of the same key is different.

### What did you expect to happen?

the three node revision is the same, the compact is normal, db size is not full.

### How can we reproduce it (as minimally and precisely as possible)?

No triggers have been found so far.
In addition, etcd does not restart during the problem include abnormal node

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
# paste output here
VERSION:
	3.4.10
$ etcdctl version
# paste output here
```
VERSION:
	3.4.10
API VERSION:
	3.4
</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

auto-compaction-mode: 'periodic'
auto-compaction-retention: '30m'
snapshot-count: 10000
heartbeat-interval: 100
election-timeout: 1000
quota-backend-bytes: 0

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

**Comment by jmhbnz** [comment]
Hey @xrkCode - Thanks for raising this issue.  Can you please clarify what you mean by `No triggers have been found so far.` Are you saying that you cannot reproduce this issue?

Please provide clear steps we can follow to reproduce the issue.

**Comment by xrkCode** [comment]
> Hey @xrkCode - Thanks for raising this issue. Can you please clarify what you mean by `No triggers have been found so far.` Are you saying that you cannot reproduce this issue?
> 
> Please provide clear steps we can follow to reproduce the issue.

Hey, Thank you for your reply.
Currently, this problem occurs only once. We are trying to construct an exception to reproduce the problem. However, the problem has not been reproduced yet.

**Comment by tjungblu** [comment]
@xrkCode I don't want to dismiss this issue, but 3.4.10 seems fairly old. Did you try upgrading to a newer version? 

IIRC #13766 was only in 3.5, #11651 might be applicable - are you using auth?
